unit-tests/src/test/java/org/eclipse/collections/impl/bimap/mutable/UnmodifiableBiMapTest.java

```diff
@@ -0,0 +1,406 @@
+    {
+        return HashBiMap.newWithKeysValues(key1, value1, key2, value2, key3, value3, key4, value4).asUnmodifiable();
+    }
+
+    @Test
+    public void newMap_throws()
+    {
+        Verify.assertThrows(IllegalArgumentException.class, () -> new UnmodifiableBiMap<>(null));
+        MutableBiMap<String, String> biMap = null;
+        Verify.assertThrows(IllegalArgumentException.class, () -> UnmodifiableBiMap.of(biMap));
+        Map<String, String> map = null;
+        Verify.assertThrows(IllegalArgumentException.class, () -> UnmodifiableBiMap.of(map));
+    }
+
+    @Override
+    @Test
+    public void containsKey()
+    {
+        MutableBiMap<Integer, Character> biMap = this.classUnderTest();
+
+        Assert.assertTrue(biMap.containsKey(1));
+        Assert.assertTrue(biMap.containsKey(null));
+        Assert.assertTrue(biMap.containsKey(3));
+        Assert.assertFalse(biMap.containsKey(4));
+    }
+
+    @Override
+    @Test
+    public void containsValue()
+    {
+        MutableBiMap<Integer, Character> biMap = this.classUnderTest();
+
+        Assert.assertTrue(biMap.containsValue(null));
+        Assert.assertTrue(biMap.containsValue('b'));
+        Assert.assertTrue(biMap.containsValue('c'));
+        Assert.assertFalse(biMap.containsValue('d'));
+    }
+
+    @Override
+    @Test
+    public void get()
+    {
+        MutableBiMap<Integer, Character> biMap = this.classUnderTest();
+        Assert.assertNull(biMap.get(1));
+        Assert.assertEquals(Character.valueOf('b'), biMap.get(null));
+        Assert.assertEquals(Character.valueOf('c'), biMap.get(3));
+        Assert.assertNull(biMap.get(4));
+    }
+
+    @Override
+    @Test
+    public void iterator()
+    {
+        MutableSet<Character> expected = UnifiedSet.newSetWith(null, 'b', 'c');
+        MutableSet<Character> actual = UnifiedSet.newSet();
+        MutableBiMap<Integer, Character> biMap = this.classUnderTest();
+        Iterator<Character> iterator = biMap.iterator();
+        Assert.assertTrue(iterator.hasNext());
+        Verify.assertThrows(UnsupportedOperationException.class, iterator::remove);
+        Verify.assertSize(3, biMap);
+        Verify.assertSize(3, biMap.inverse());
+        for (int i = 0; i < 3; i++)
+        {
+            Assert.assertTrue(iterator.hasNext());
+            actual.add(iterator.next());
+        }
+        Assert.assertEquals(expected, actual);
+        Assert.assertFalse(iterator.hasNext());
+        Verify.assertThrows(NoSuchElementException.class, (Runnable) iterator::next);
+    }
+
+    @Test
+    public void testClone()
+    {
+        MutableBiMap<Object, Object> biMap = this.newMap();
+        MutableBiMap<Object, Object> clone = biMap.clone();
+        Assert.assertSame(biMap, clone);
+    }
+
+    @Test
+    public void withKeyValue()
+    {
+        Verify.assertThrows(UnsupportedOperationException.class, () -> this.getEmptyMap().withKeyValue(1, 'a'));
+    }
+
+    @Override
```

Comment:
In this case @Test is missing. Same case for couple of other methods in this class.